src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -184,20 +184,29 @@ private void DetermineIgnoreCondition(JsonIgnoreCondition? ignoreCondition, bool
 
         private void DetermineNumberHandling(JsonNumberHandling? parentTypeNumberHandling)
         {
+            bool propertyIsNumberOrNumberCollection = ConverterBase.IsInternalConverterForNumberType || TypeIsCollectionOfNumbers();
```

Comment:
Since this is only used in the line below, you could collapse it. That will prevent the somewhat expensive `TypeIsCollectionOfNumbers()` from running when it isn't necessary.